src/Tools/ExternalAccess/Razor/RazorAnalyzerAssemblyResolver.cs

```diff
@@ -54,6 +56,16 @@ internal sealed class RazorAnalyzerAssemblyResolver() : IAnalyzerAssemblyResolve
             }
 
             var assemblyFileName = $"{assemblyName.Name}.dll";
+
+            // Depending on who wins the race to load these assemblies, the base directory will either be the tooling root (if Roslyn wins)
+            // or the ServiceHubCore subfolder (razor). In the root directory these are netstandard2.0 targeted, in ServiceHubCore they are 
+            // .net targeted. We need to always pick the same set of assemblies regardless of who causes us to load. Because this code only
+            // runs in a .net based host, it's safe to always choose the .net targeted ServiceHubCore versions.
+            if (!directory.TrimEnd('/', '\\').EndsWith(ServiceHubCoreFolderName))
```

Comment:
nit: I think we could use spans to avoid allocations, something like the following.

Also looks like default for EndsWith is CurrentCulture which we probably don't want. And if we are on windows, presumably we want to ignore casing?

```suggestion
            if (!directory.AsSpan().TrimEnd('/', '\\').EndsWith(ServiceHubCoreFolderName, StringComparison.OrdinalIgnoreCase))
```